mozilla/telemetry-airflow
# Airflow 1.10.7 upgrade

TODO:
- [x] fix initial parse issues with all dags

Changelogs:
- [x] go over changelogs. Looks like these are of note, although we don't rely on most of these:
- add envFrom for Kubernetes Executor
- KubernetesPodOperator support for envFrom configMapRef/secretRef. Also some improvements with display of pod resources, etc.
- Dataflow hook pinned to py2. Ability to add labels.
- GoogleCloudStorageToBigQueryOperator will be renamed GCSToBigQueryOperator in the future (not in 1.10.7). Also had some fixes (autodetect, encoding param, etc.).
- Add support CSV files in MySqlToGoogleCloudStorageOperator
- Add run_as_user and fs_group options for Kubernetes
- [x] Changes to Google Cloud Transfer Service Operators - e.g. changes to s3_to_gcs_transfer_operator.py which we rely on (https://github.com/apache/airflow/pull/4792), timeout parameter added, moved source to airflow.contrib.operators.gcp_transfer_operator
- Add Google Cloud Translate operator (do we have a use case for this?)
- Operators for S3 object copying/deleting
- Add compose to GoogleCloudStorageHook (not used currently)
- make number of dataproc masters configurable
-  Add more metrics around the scheduler (airflow scheduler, although we don't scrape metrics in a prom fashion, and google doesnt either?)
- [x]  Use GOOGLE_APPLICATION_CREDENTIALS constant from library (https://github.com/apache/airflow/pull/5744) - We fork GKEPodOperator to set CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE to use service account key jsons to issue gcloud commands, plus some xcom changes
- [x] Fix Databricks hook leaks API secret in logs - We only use the token for local dev testing. bin/run reflects the change. Prod jobs use the WTMO connection which uses basic user/pass auth
- KubernetesPodOperator fixes and test
- Environment variables not correctly set in SparkSubmitHook 
- [x] Add ability to specify optional components in DataprocClusterCreateOperator (we were doing this via a fork, so eliminate duplicate code where possible) - https://github.com/apache/airflow/pull/5821/files


review code for the following between 1.10.2 and 1.10.7
- [x] dags/operators/gcp_container_operator (diff 1.10.2 and 1.10.7, copypasta the 1.10.7 one into our own fork and add our changes again (https://github.com/mozilla/telemetry-airflow/blame/master/dags/operators/gcp_container_operator.py). This functionality will migrate with newer versions as this is deprecated in master.
- [x] dags/utils/dataproc.py -> check bc we import the dataproc operators from 1.10.2 (i know that optional components is now supported for example). pysparkop changes look ok


- [x] dags/operators/gcs.py -> https://github.com/apache/airflow/blob/a87f3dfc0c513b05271cba7051c18e4e742ef245/airflow/gcp/operators/gcs.py#L321 + changes for 1.10.2 /py2 , might be able to delete
- [x] dags/operators/moz_dataproc_operator.py -> can we delete this now? -> no bc we use component gateway. updated all refs
- [x] git grep s3_to_gcs_transfer_operator - there are changes in (https://github.com/apache/airflow/pull/4792)
- [x] documentation  - update readme on how to troubleshoot test-parse environment circleci failures
- [x] rm dags/operators/dataproc_hadoop_with_aws.py - no longer needed

Test:
- [x] GkePodOperator - clean gke pods should suffice
- [x] DataprocOperator - verify component gateway installed on new cluster
- [x] DataprocOperator - test pyspark runner
- [x] DataprocOpertaor - test jar runner
- [x] DataprocOperator - test scriptrunner
- [x] dags/utils/dataproc.py -> Test dataproc runners without changing main_class to main_jar (this was a bug found in earlier dataproc testing, that may have been fixed) for jar and script runner.
- [x] test s3togcs transfer job operators
- [x] dags/utils/gcp.py -> also imports dataproc operators and gkepodoperator, also some bigquery operators.  git grep utils.gcp for a list of dags using pieces of this. Assuming the dags load, and diffs don't look bad, rely on devs to keep an eye on these after upgrade
- [x] utils/amplitude - some bqtocloudstoageoperators. Rely on devs to keep an eye on. Diffs are minimal also

Other
- Makefile clean target cleans out local logs
- Add timeout to s3togcs operators because it now defaults to 60s.
- Add common defaults to our GkePodOperator to simplify usage, and rm'd usage from dags
- Add pod name label (without the uuid) to pod
- Using upstream gcs_delete_operator and removing our version
- Random other changes to reflect upstream changes
